common/persistence/metered.go

```diff
@@ -67,6 +68,209 @@ func (r GetAllHistoryTreeBranchesResponse) Len() int {
 	return len(r.Branches)
 }
 
+// For responses that require metrics for payload size EstimatePayloadSizeInBytes() int should be defined.
+
+func (r *GetReplicationTasksResponse) EstimatePayloadSizeInBytes() int {
```

Comment:
I think @timl3136 is introducing a `Size` interface for this that is probably worth adopting